sdks/python/apache_beam/internal/cloudpickle_pickler.py

```diff
@@ -99,6 +103,14 @@ def _create_absl_flags():
   return flags.FLAGS
 
 
+def _pickle_rlock(obj):
+  return _create_rlock, tuple([])
+
+
+def _create_rlock():
+  return RLOCK_TYPE()
```

Comment:
Should this return RLOCK_TYPE instead of RLOCK_TYPE() ?

Reply:
Also, consider adding a test.

Reply:
I added two tests. One for the default lib (dill) and one for cloudpickle.

Also I want to return RLOCK_TYPE() which makes a new instance of the type.  Dill does it similarily here:
https://github.com/uqfoundation/dill/blob/04aba03432955f64c05441ffe0553af987c087ff/dill/_dill.py#L744

What dill does that doesn't make any sense for cloud pickle is it also tries to populate whether or not the lock is locked. That might make sense for save session but makes no sense when it is just a class

Reply:
Ok. perhaps I was asking a wrong question. I was looking at
https://github.com/cloudpipe/cloudpickle/blob/5d89947288a18029672596a4d719093cc6d5a412/cloudpickle/cloudpickle_fast.py#L528, and it seemed to me that the reduce function (e.g. `_mappingproxy_reduce`) returns a tuple: Type object + Arguments.

While in your case `_pickle_rlock` returns an instance of the type (i.e. actual RLock) + arguments.

Reply:
I actually looked at this a bit more. The reduce function can return a (type, parameters) or (function, parameters)

I was using the (function, parameters) but upon experimentation I realized I could simplify this by just using the type.